onnxruntime/core/util/thread_utils.cc

```diff
@@ -144,4 +169,15 @@ ORT_API_STATUS_IMPL(SetGlobalCustomJoinThreadFn, _Inout_ OrtThreadingOptions* tp
   return nullptr;
 }
 
+ORT_API_STATUS_IMPL(SetGlobalIntraOpThreadAffinity, _Inout_ OrtThreadingOptions* tp_options, const char* affinity_string) {
+  if (!tp_options) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "Received null OrtThreadingOptions");
+  }
+  if (!affinity_string) {
```

Comment:
what happens if the string is empty?

Reply:
Condition removed, allowing an empty string to be set as an affinity string, meaning no affinity.

Reply:
An empty string could be an honest mistake by the user. We should return an error instead of silently allowing it.

Reply:
no exactly - in production user may read the affinity string from a .cfg file, which might be empty depending on the hardware to their discretion.